sdk/tests/iot/common/test_az_iot_common.c

```diff
@@ -258,20 +258,6 @@ static void test_az_iot_provisioning_client_logging_succeed()
   az_log_set_classifications(NULL);
 }
 
-static void test_az_log_incomplete_list_fails_gracefully()
-{
-  az_log_classification const classifications[] = { AZ_LOG_HTTP_REQUEST };
```

Comment:
Please write a test case covering that the precondition gets triggered when you pass an array of `{AZ_LOG_HTTP_REQUEST, (az_log_classification)12345}` (run it only when built with preconditions on).
There's a way to verify preconditions triggered, see some other tests. If you don't want to invest in writing that test until we held a broader discussion on preventing incorrect behavior on incorrect input data - that's fine, just please don't forget about it, create a work item to track if neccessary.

Reply:
I'll punt this for now, but will keep track of it and follow up as part of the design issue.